### PR TITLE
add upgrade mainnet identity

### PR DESCRIPTION
**Pull request by waleedhammam**
### Changes

- auto upgrade mainnet
- save explorers in js-configs to reuse later
- After migration is complete we will trigger this update by setting new config `has_migrated=True`

### Related Issues

- https://github.com/threefoldtech/js-sdk/issues/1071

### CODE REVIEW

File: jumpscale/core/identity/__init__.py [hunk 1]
```diff
@@ -87,6 +98,13 @@ def explorer(self):
             from jumpscale.clients.explorer import export_module_as  # Import here to avoid circular imports
 
             ex_factory = export_module_as()
+
+            # Backward compitablity (Update mainnet url)
+            # Create new config has_migrated = True after mainnet update
+            if js_config.get("has_migrated", False):
```
Comment (abom) [inline jumpscale/core/identity/__init__.py:104]: Can we rename it to a more specific name, to be related to this explorer URL change.
Also, we can add a note about better network selection implementation.

File: jumpscale/core/identity/__init__.py [hunk 2]
```diff
@@ -87,6 +98,13 @@ def explorer(self):
             from jumpscale.clients.explorer import export_module_as  # Import here to avoid circular imports
 
             ex_factory = export_module_as()
+
+            # Backward compitablity (Update mainnet url)
+            # Create new config has_migrated = True after mainnet update
+            if js_config.get("has_migrated_explorer_url", False):
+                if urlparse(self.explorer_url).hostname == urlparse(EXPLORER_URLS["mainnet"]).hostname:
+                    self.explorer_url = EXPLORER_URLS["mainnet"]
```
Comment (abom) [inline jumpscale/core/identity/__init__.py:106]: We should set `has_migrated_explorer_url` to `True` here, so we don't do it again.

File: jumpscale/core/identity/__init__.py [hunk 3]
```diff
@@ -87,6 +98,13 @@ def explorer(self):
             from jumpscale.clients.explorer import export_module_as  # Import here to avoid circular imports
 
             ex_factory = export_module_as()
+
+            # Backward compitablity (Update mainnet url)
+            # Create new config has_migrated = True after mainnet update
+            if js_config.get("has_migrated_explorer_url", False):
```
Comment (abom) [inline jumpscale/core/identity/__init__.py:104]: This should be `if not...`

**Comment by waleedhammam** [comment]
TO MERGE AFTER UPGRADING MAINNET URL

**Review by xmonader — APPROVED** [review]

**Pull request merged**